<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>
<body>
  <script>
    const promise = new Promise(function (resolve, reject) {
      resolve('ok');
      setTimeout(function () { throw new Error('test') }, 3000)
    });
    promise.then( value => {
      console.log(value) 
    }).catch( error => { // 在then运行后，catch仍可以捕捉到3秒后抛出的错误
      console.log(error);
    })
  </script>
</body>
</html>
